### PR TITLE
feat(trading-wallet): implemented send to trading wallet for BTC

### PR DESCRIPTION
**Pull request by milan-bc**
## Description (optional)
This PR cover sending for BTC, more PRs about to be come. 

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/types.ts [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+    name: string | null
+    recipient: string | null
+    recipientAddress: string | null
+    routingNumber: string | null
+    swiftCode: string | null
+  }
+  currency: CoinTypeEnum
+  id: string
+  state: 'PENDING' | 'ACTIVE'
+}
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/types.ts:20]: There is already a type for this called `BeneficiaryType`

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts [hunk 2]
```diff
@@ -71,6 +72,10 @@ export default ({
       })
       payment = yield payment.init()
       let defaultAccountR
+
+      const account: AccountType = yield call(api.getSBPaymentAccount, 'BTC')
+      yield put(A.sendBtcAccountsSuccess(account))
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:77]: I would much rather do all of this in the send/sagas.ts, similar to exchangePaymentAccount

https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/philip/wal-61-buy-ui-clean-up/packages/blockchain-wallet-v4-frontend/src/data/components/send/reducers.ts/#L6-L14

**Review by plondon — CHANGES_REQUESTED** [review]
Reuse BeneficiaryType

Change references of "account" to "tradingAccount"

Move api call, saga, reducer, selector to send/ (not coin specific)

**Comment by jovi5513** [comment]
8

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/types.ts (hunk 1 above)
Comment (jovi5513) [reply packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/types.ts:20]: 1

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts (hunk 2 above)
Comment (jovi5513) [reply packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:77]: 2

**Review by jovi5513 — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts [hunk 3]
```diff
@@ -6,3 +6,6 @@ export const FETCH_PAYMENTS_ACCOUNT_EXCHANGE_FAILURE =
   '@EVENT.SEND.FETCH_PAYMENTS_ACCOUNT_EXCHANGE_FAILURE'
 export const FETCH_PAYMENTS_ACCOUNT_EXCHANGE_SUCCESS =
   '@EVENT.SEND.FETCH_PAYMENTS_ACCOUNT_EXCHANGE_SUCCESS'
+
+export const FETCH_PAYMENTS_TRADING_AACCOUNTS_SUCCESS =
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts:10]: small misspelling

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts:10]: Any reason to only have SUCCESS actionType?

Usually in this scenario we would have 4:
1. actionType to kick off a saga
2. SUCCESS
3. FAILURE
4. LOADING

File: packages/blockchain-wallet-v4-frontend/src/data/components/send/types.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
-import { CoinType, RemoteDataType } from 'core/types'
+import { BeneficiaryType, CoinType, RemoteDataType } from 'core/types'
 
 export type SendState = {
   exchangePaymentsAccount: {
     [key in CoinType]: RemoteDataType<any, any>
   }
+  tradingPaymentsAccount: {
+    [key in CoinType]: RemoteDataType<any, BeneficiaryType>
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/send/types.ts:8]: 👍

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts [hunk 5]
```diff
@@ -65,12 +65,16 @@ export default ({
         payPro
       } = action.payload
       yield put(A.sendBtcPaymentUpdatedLoading())
+
+      // eslint-disable-next-line
+      console.log('idemoooo')
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBtc/sagas.ts:70]: Remove

**Review by plondon — CHANGES_REQUESTED** [review]
Small changes left!

File: packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts (hunk 3 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts:10]: can add others of course, didn't sow benefit to use them but I'll follow up standards

Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/data/components/send/actionTypes.ts:10]: How do you handle the case of the the request failing for instance?

File: packages/blockchain-wallet-v4-frontend/src/data/components/send/selectors.ts [hunk 6]
```diff
@@ -9,3 +9,11 @@ export const getPaymentsAccountExchange = curry(
     )
   }
 )
+
+export const getPaymentsTradingAccountAddress = curry(
+  (currency: CoinType, state: RootState) => {
+    return state.components.send.tradingPaymentsAccount[currency].map(
+      x => x.address
+    ).data
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/send/selectors.ts:17]: `.data` should **NEVER** be used, _unless_ you know **FOR SURE** the data is **Success**ful

**Comment by plondon** [comment]
A non nabu user should not be able to PUT a payments/accounts/simplebuy, and as you can see the request fails. However we are still showing it as an option:

Screen Shot 2020-09-01 at 1 52 25 PM

**Comment by plondon** [comment]
```
const accountAddress = hasAccountAddress
    ? accountAddressR.data
    : Remote.of('')
```
I believe this is the issue.

**Review by plondon — APPROVED** [review]

**Pull request merged**